### PR TITLE
Publish artifacts for docs.ms

### PR DESCRIPTION
**Pull request by danieljurek**

### CODE REVIEW

File: eng/scripts/New-DocsMsArtifact.ps1 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+npm i -g moxygen
+
+# Moxygen will fail if docs/docs.ms does not exist
+mkdir -p docs/docs.ms
+
+moxygen --anchors --output docs/docs.ms/api-docs.md docs/xml
```
Comment (scbedd) [inline eng/scripts/New-DocsMsArtifact.ps1:6]: Will this run per _artifact_ (eg `storage`) or will it run for the entire repo?

Comment (danieljurek) [reply eng/scripts/New-DocsMsArtifact.ps1:6]: In C this runs once for the entire repo. For C++ I am planning to have it run once per-package.

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 2]
```diff
@@ -398,3 +417,8 @@ jobs:
       inputs:
         artifactName: docs
         path: $(Build.ArtifactStagingDirectory)/docs
+
+    - task: PublishPipelineArtifact@1
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:421]: @scbedd after our last conversation any thoughts on how we might want to structure the artifact so there is only one artifact per package?

Comment (danieljurek) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:421]: In C this is not a problem. In C++ I nest these within folders: https://github.com/Azure/azure-sdk-for-cpp/pull/576

File: doc/Doxyfile [hunk 3]
```diff
@@ -1998,7 +1998,7 @@ MAN_LINKS              = NO
 # captures the structure of the code including all documentation.
 # The default value is: NO.
 
-GENERATE_XML           = NO
+GENERATE_XML           = YES
```
Comment (ahsonkhan) [inline doc/Doxyfile:2001]: Why was this needed?
